src/Peachpie.Library.Graphics/PhpGd2.cs

```diff
@@ -1169,5 +1159,831 @@ static bool imagesave(Context ctx, PhpResource im, PhpValue to/* = null*/, Actio
             return true;
         }
 
+        #endregion imagesave
+
+        // NEW FUNCTIONS
+
+        #region imagefilter
+
+        /// <summary>
+        /// Applies a filter to an image
+        /// </summary>
+        public static bool imagefilter(PhpResource src_im, int filtertype) => imagefilter(src_im, filtertype, -1, -1, -1, -1);
```

Comment:
unnecessary to declare so many overrides,. Use default values instead., This slows down runtime significantly.